gradle/lib/dependabot/gradle/file_updater.rb

```diff
@@ -69,6 +68,13 @@ def update_buildfiles_for_dependency(buildfiles:, dependency:)
 
           buildfile = files.find { |f| f.name == new_req.fetch(:file) }
 
+          # Exception raised to handle issue that arises when buildfiles function (see this file)
+          # removes the build file that contains the dependency itself. So no build file exists to
+          # update dependency, This behaviour is evident for extremely small number of users
+          # that have added separate repos as sub-modules in parent projects
+
+          raise DependencyFileNotResolvable, "No build file found to update the dependency" if buildfile.nil?
```

Comment:
@abdulapopoola , Let me check on this and come back to you.

Reply:
@abdulapopoola ,
This issue is specific to Gradle only because Sentry logged it considering the error context (throw position, variable context)
I need to check what approach do we have for sub modules. Might be only Gradle parser is having this issue. Haven't checked for other eco systems. Let me check on it and come back to you.

Reply:
@sachin-sandhu , you're right, Sentry also only shows [gradle](https://github.sentry.io/discover/results/?field=title&field=release&field=environment&field=user.display&field=timestamp&field=gh.dependabot_api.update_config.package_manager&name=NoMethodError%3A+undefined+method+%60content%27+for+nil&project=1451818&query=issue%3ADELTAFORCE-Z59+%21gh.dependabot_api.update_config.package_manager%3A%22%22&sort=gh.dependabot_api.update_config.package_manager&statsPeriod=90d&yAxis=count%28%29).

It does seem that the error has stopped now but it's still good to add in the more specific error so that if it recurs, we can quickly root cause it.

Reply:
@abdulapopoola , Gave it another thought. I think instead of throwing exception which will end up in Sentry again, Another way would be to just exclude the update for particular dependency and let update continue for others.

Reply:
Makes a lot of sense since we do a best effort approach, I would hope we write the logs about why this update fails (e.g. for grouped update scenarios). I also still think we should have a proper error class for this (e.g. SubModulesNotConfigured or something similar).

Reply:
> Another way would be to just exclude the update for particular dependency and let update continue for others.

It's a nice idea, but if we can't pull the dependency, than how can we safely walk the tree? Like the unreachable dependency may have a version pin on another dependency in the tree that prevents updating it.

I know we've had similar requests in Core of "please ignore my private deps that are on some private registry" and we invariably end up realizing without the full tree we can't propose any updates with confidence.

Reply:
Also, since `DependencyFileNotResolvable` is a well-known error which we classify as "problem with the user's repo" and not "our internal code error", it won't show up in Sentry again for us...

Here it is arguably a code issue with Dependabot, although I'd say it's an edge case missing feature rather than an out and out bug...